src/afl-fuzz.c

```diff
@@ -1288,13 +1306,47 @@ int main(int argc, char **argv_orig, char **envp) {
 
       /* afl-qemu-trace takes care of converting AFL_PRELOAD. */
 
+    } else if (afl->fsrv.frida_mode) {
+
+      afl_preload = getenv("AFL_PRELOAD");
+      frida_afl_preload_reqd =
+          snprintf(NULL, 0, "%s %s", afl_preload, "./afl-frida-trace");
+      if (frida_afl_preload_reqd < 0) {
+
+        FATAL("Invalid AFL_PRELOAD length: %d", frida_afl_preload_reqd);
+
+      }
+
+      frida_afl_preload = malloc(frida_afl_preload_reqd + 1);
+      frida_afl_preload_len =
+          snprintf(frida_afl_preload, frida_afl_preload_reqd + 1, "%s %s",
+                   afl_preload, "./afl-frida-trace");
+
+      if (frida_afl_preload_len != frida_afl_preload_reqd) {
+
+        FATAL("Invalid AFL_PRELOAD length: %d != %d", frida_afl_preload_len,
+              frida_afl_preload_reqd);
+
+      }
+
```

Comment:
getenv can return NULL, afl-frida-trace is most likely not the in the current directory plus this is overly complicated.
simply do:
```
...
afl_preload = getenv("AFL_PRELOAD");
u8 *frida_binary = find_binary("afl-frida-trace");
if (afl_preload) { frida_afl_preload = alloc_printf("%s:%s", afl_preload, frida_binary); }
else { frida_afl_preload = alloc_printf("%s", frida_binary); }
OKF("Frida Mode setting LD_PRELOAD %s", frida_afl_preload);
...
```

Reply:
Yeah. Thanks, that code was a bit clumsy.

Reply:
find_binary can now be switched to find_afl_binary if you pull into your repo

Reply:
Fixed.